packages/nextjs/app/api/configure-contract/route.ts

```diff
@@ -0,0 +1,36 @@
+import { NextRequest, NextResponse } from "next/server";
+import path from "path";
+import prettier from "prettier";
+import fs from "fs";
+
+export async function POST(req: NextRequest) {
+  const TARGET_DIR = path.join(process.cwd(), "contracts");
+  try {
+    const data = await req.json();
+
+    const generatedContractComment = `/**
+* This file is autogenerated by Scaffold-Stark.
+* You should not edit it manually or your changes might be overwritten.
+*/`;
+
+    const fileContent = await prettier.format(
+      `${generatedContractComment}\n\nconst configExternalContracts = ${JSON.stringify(data)} as const;\n\nexport default configExternalContracts;`,
+      {
+        parser: "typescript",
+      },
+    );
+
+    fs.writeFileSync(
```

Comment:
Will it be used in `Vercel`, from my understanding, `Vercel` uses a serverless function (not a traditional server) to serve files. `fs.writeFileSync` may fail in production.

Reply:
@iossocket i dont think this page is supposed to go to production at all. its a helper tool for the users to debug existing deployed contracts that they didnt do themselves. But i agree with @iossocket on the usage here, was wondering why is the file generation not done in client side?

Reply:
This feature works very well, and facilitates automatic download and upload without having to replace the file, but as the guys @metalboyrick  and @iossocket say, I don't know if it would be optimal or if it could fail,

Could we have this version from the client side???

Reply:
the idea was to make user convenient without the need to replace the file, but considering production, i think we can have this feature for localhost, and file generation for production?

Reply:
IMO, this would be a good option.

- Automatic file upload operation localhost
- File download and replacement in production.

Reply:
@metalboyrick @iossocket what do you think

Reply:
i meant that in production, people wouldnt need this feature since the application would be used by consumer not the devs haha, devs would only need this for local or staging environments to extract contract ABIs from the network.

I was thinking, we CAN actually have server-side API (which one works then we just go lah, don't think its a big deal) , in the local, but we need to ensure (or at least have a guide on this in the docs) that the devs that will be using this feature will be able to shut it off somehow (if server), because server cost can rack up too in vercel haha, and vercel is famous for being very expensive

cc @iossocket 

p.s. pls ping me on slack or tele if I dont respond in a while ser, github notifications get very cluttered very fast haha, sorry for the late reply

Reply:
@iossocket 👀